tests/test_thunderbird.py

```diff
@@ -349,10 +349,22 @@ def send_email(tb, sign=False, encrypt=False, inline=False, attachment=None):
                 button('Protect subject').doActionNamed('press')
     except tree.SearchError:
         pass
+    # if there is a dialog window, then there must have been some issue sending
+    try:
+        dialog = tb.app.dialog('.*')
+        try:
+            dialog_text = dialog.child(roleName='label').text
+            raise Exception("Failed to send message with error '{}'"\
+                .format(dialog_text))
```

Comment:
Ouch, there are more possible dialogs:
```
Exception: Failed to send message with error 'Status:'
```
One way is to wait (with a loop?) until no dialog is found or some timeout expires (10s?) - in the case of a timeout, report the most recent dialog found.

Reply:
Does this not catch that dialog? How can I make it give that dialog?

Reply:
The problem is it catches _any_ dialog as a failure. This one is just showing a progress of sending the email...

Reply:
This solution even though more efficient was more error-prone. I'll go another way.